README.rst

```diff
@@ -97,6 +97,23 @@ Ideas that we already did
 Recent Changes
 ==============
 
+- 0.4.0
+
+  - This is the last release series for this ``ordermap`` under that name,
+    because the crate is **going to be renamed** to ``indexmap`` (with types
+    ``IndexMap``, ``IndexSet``) and no change in functionality!
+  - The map and its associated structs moved into the ``map`` submodule of the
+    crate, so that the map and set are symmetric
+
+    + The iterators, ``Entry`` and other structs are now under ``ordermap::map::``
+
+  - Internally refactored ``OrderMap<K, V, S>`` so that all the main algorithms
+    (insertion, lookup, removal etc) that don't use the ``S`` parameter (the
+    hasher), are compiled without depending on ``S``.
```

Comment:
Does this have any user-visible impact?  I think it *could* if all such methods were in `impl` without `S: BuildHasher`, as some methods already are, but I don't know if this is very meaningful.  Besides that, I guess the benefit is reduced monomorphization for these implementations?

Reply:
Yep, it's intended to reduce generics bloat, even if I guess it might not make the difference in most use cases.

`Entry<K, V, S>` -> `Entry<K, V>` is the user visible thing I can think of (and the component structs of entries). There should be no other visible changes. The refactoring was surprisingly smooth, but I suppose the diff is not so easy to read.  ( `git diff --stat -p 324caad..b159fad` )

That reminds me, I didn't visit OrderSet at all for that refactoring, but I think most of its unique features (like subset / difference) depend on the hasher.

Reply:
There's even the closure to `Bucket::key_ref` accessors (and similar) refactoring in there. The main motivation was to reuse the same function instead of having a separate identical closure for each iterator method in the same iterator.

Reply:
Yeah, most of the new set stuff uses `contains(item)`.

Maybe just add a final clause here: "without depending on `S`, reducing generics bloat."

Reply:
amended with that, thanks!